test/binary-parser.test.js

```diff
@@ -217,8 +214,20 @@ function fieldParsingTests () {
       expect(field.type.name).toEqual(f.type_name)
     })
   })
+  test("Field throws when type code out of range", () => {
+    const parser = makeParser("0101");
+    expect(() => parser.readField()).toThrow(new Error("Cannot read FieldOrdinal, type_code out of range"));
```

Comment:
In typescript you can extend `Error` which will make it so you don't need to hardcode these values. 

This might be a good area for future improvement. You could either have a generic error (like `CodecError`) that has an enum based code and a message and check on enum, or you could build a specific error like `FieldOutOfRangeError`.

Reply:
Agreed to look at this in a follow up PR based around better error handling and using: https://github.com/goldbergyoni/nodebestpractices/blob/master/sections/errorhandling/useonlythebuiltinerror.md